channeldb/payment_control.go

```diff
@@ -311,15 +311,12 @@ func (p *PaymentControl) RegisterAttempt(paymentHash lntypes.Hash,
 		}
 
 		// We cannot register a new attempt if the payment already has
-		// reached a terminal condition. We check this before
-		// ensureInFlight because it is a more general check.
-		settle, fail := p.TerminalInfo()
-		if settle != nil || fail != nil {
-			return ErrPaymentTerminal
-		}
+		// reached a terminal condition.
```

Comment:
nit: commit message mentions new unit test, but that's only in a later commit.

Reply:
fixed!